resources/models.py

```diff
@@ -451,6 +453,45 @@ class Meta:
         abstract = True
 
 
+class BerthManager(models.Manager):
+    def get_queryset(self):
+        """
+        The QuerySet annotates whether a berth is available or not. For this,
+        it considers the following criteria:
+            - If there are leases associated to the berth
+            - If any lease ends during the current or the last season (previous year)
+                + If a lease ends during the current season:
+                    * It needs to have a "valid" status (DRAFTED, OFFERED, PAID)
+                + If a lease ended during the last season:
+                    * It needs to have a "valid" status (PAID)
+                    * It needs to have renew automatically set
+        """
+        from leases.models import BerthLease
+
+        current_season_end = calculate_berth_lease_end_date()
+
+        in_current_season = Q(end_date=current_season_end)
+        in_last_season = Q(
+            end_date=current_season_end.replace(current_season_end.year - 1)
+        )
+        active_current_status = Q(
+            status__in=(LeaseStatus.DRAFTED, LeaseStatus.OFFERED, LeaseStatus.PAID)
+        )
+        paid_status = Q(status=LeaseStatus.PAID)
+        auto_renew = Q(renew_automatically=True)
+
+        active_leases = (
+            BerthLease.objects.filter(berth=OuterRef("pk"))
+            .filter(
+                Q(in_current_season & active_current_status)
+                | Q(in_last_season & auto_renew & paid_status)
+            )
+            .values("id")
```

Comment:
hm, I think this is not needed, since it fires the `EXISTS NOT ... ` query, which does not care about the values. at least, when tested locally, I could not see any difference, if I tested it correctly of course

Reply:
Yeah, it doesn't seem to have any performance improvement. However, it simplifies a bit the SQL generated by that QuerySet, removing some unncessary fields. We can remove it anyway since it's no affecting direclty and we (thankfully) don't get to see the raw SQL often.

```sql
NOT EXISTS(SELECT
        U0."id",
        U0."created_at",
        U0."modified_at",
        U0."customer_id",
        U0."boat_id",
        U0."status",
        U0."comment",
        U0."berth_id",
        U0."application_id",
        U0."start_date",
        U0."end_date",
        U0."renew_automatically" 
    FROM
        "leases_berthlease" U0 
    WHERE
        (
            U0."berth_id" = (
                "resources_berth"."id"
            ) 
            AND (
                (
                    U0."end_date" = 2020-09-14 
                    AND U0."status" IN (
                        drafted, offered, paid
                    )
                ) 
                OR (
                    U0."end_date" = 2019-09-14 
                    AND U0."renew_automatically" = True 
                    AND U0."status" = paid
                )
            )
        )) AS "is_available"
```

```sql
NOT EXISTS(SELECT
        U0."id" 
    FROM
        "leases_berthlease" U0 
    WHERE
        (
            U0."berth_id" = (
                "resources_berth"."id"
            ) 
            AND (
                (
                    U0."end_date" = 2020-09-14 
                    AND U0."status" IN (
                        drafted, offered, paid
                    )
                ) 
                OR (
                    U0."end_date" = 2019-09-14 
                    AND U0."renew_automatically" = True 
                    AND U0."status" = paid
                )
            )
        )) AS "is_available"
```

Reply:
I think at the end of the day postgres does not care for columns to be selected in the `SELECT` subquery. Deleting `.values("id")` however simplifies our python code (although just by a bit 😅)